### PR TITLE
Upgrade to Spark 2.0.0 and Scala 2.11; add DynamoDB Maven plugin

### PR DESCRIPTION
**Pull request by timchan-lumoslabs**
I added this plugin: http://dynamodb.jcabi.com/ 

The integration test that you currently have now runs under `mvn integration-test` instead of `mvn test`.

### CODE REVIEW

File: src/main/scala/com/github/traviscrawford/spark/dynamodb/DynamoDBRelation.scala [hunk 1]
```diff
@@ -166,3 +166,7 @@ private case class ScanConfig(
   maybeCredentials: Option[String],
   maybeRegion: Option[String],
   maybeEndpoint: Option[String])
+
+object Logger extends Serializable {
+  @transient lazy val log = org.apache.log4j.LogManager.getLogger(DynamoDBRelation.getClass())
```
Comment (traviscrawford) [inline src/main/scala/com/github/traviscrawford/spark/dynamodb/DynamoDBRelation.scala]: Let's use `slf4j` like Spark does:

```
import org.slf4j.LoggerFactory

@transient lazy val log = LoggerFactory.getLogger(DynamoDBRelation.getClass)
```

File: src/test/scala/com/github/traviscrawford/spark/dynamodb/DynamoDBRelationIntegrationSpec.scala [hunk 2]
```diff
@@ -1,10 +1,8 @@
 package com.github.traviscrawford.spark.dynamodb
 
 import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient
-import com.amazonaws.services.dynamodbv2.document.DynamoDB
-import com.amazonaws.services.dynamodbv2.document.Item
+import com.amazonaws.services.dynamodbv2.document.{DynamoDB, Item}
```
Comment (traviscrawford) [inline src/test/scala/com/github/traviscrawford/spark/dynamodb/DynamoDBRelationIntegrationSpec.scala]: Let's revert this change - my preference is importing one item per line.

**Comment by timchan-lumoslabs** [comment]
@traviscrawford my last commit has the changes you requested.

**Comment by findchris** [comment]
@traviscrawford @timchan-lumoslabs - What's the status here?  Can I help with any changes to make this Spark 2.0-compatible?  We can still use `DataFrame`s in Spark 2.0, so the changes shouldn't be too significant, and I'm happy to help.

More color:  I am upgrading to Spark 2.0, and came across this error:
```
java.lang.NoSuchMethodError: com.github.traviscrawford.spark.dynamodb.package$DynamoDBDataFrameReader.dynamodb(Ljava/lang/String;)Lorg/apache/spark/sql/Dataset;
```

**Comment by findchris** [comment]
Btw, I think the above issue arises from a change to `DataFrameReader` from Spark 1.6 to Spark 2.0.

In Spark 1.6, `DataFrameReader.load` [returns a `DataFrame`](https://spark.apache.org/docs/1.6.1/api/java/org/apache/spark/sql/DataFrameReader.html#load()), whereas in Spark 2.0, [it returns a `DataSet<Row>`](https://spark.apache.org/docs/2.0.0/api/java/org/apache/spark/sql/DataFrameReader.html#load()).  

I think the solution is to simply change the return type of `DynamoDBDataFrameReader.dynamodb` from `DataFrame` to `DataSet<Row>`.  From [the `DataSet` docs](https://spark.apache.org/docs/2.0.0/api/java/org/apache/spark/sql/Dataset.html:
> Each Dataset also has an untyped view called a DataFrame, which is a Dataset of Row.
It seems that the interface should be backwards compatible with the `DataFrame` API, so hopefully the scope is small.

Is this PR active?

**Comment by timchan-lumoslabs** [comment]
@findchris I'm not quite sure why this PR wasn't merged. I'd be willing to update this PR if @traviscrawford would like to upgrade to Spark 2.0.

**Comment by traviscrawford** [comment]
Hi @timchan-lumoslabs - this is a good change, and if you'd like to update against the current master let's go ahead and merge. Largely I've been dragging my feet on this waiting for a Spark point release to come out before upgrading internally, and with 2.0.1 out let's go ahead and get this merged.

cc @sboora

**Comment by findchris** [comment]
@traviscrawford @timchan-lumoslabs - Please note the incompatibility I mentioned in the [above comment](https://github.com/traviscrawford/spark-dynamodb/pull/8#issuecomment-252777066), and let me know if you need a hand with anything.

**Comment by findchris** [comment]
@timchan-lumoslabs Can you rebase to `traviscrawford/spark-dynamodb`'s `master` and incorporate feedback from @traviscrawford?  I can try your feature branch out with Spark 2.0 in production.  I'd love to get this merged in, as Spark 2.0 is alive and well!

**Comment by findchris** [comment]
Hey @traviscrawford @timchan-lumoslabs - Any movement here?

**Comment by timchan-lumoslabs** [comment]
@findchris I was planning on working on this over the weekend. Sorry about the delay.

**Comment by findchris** [comment]
Thanks @timchan-lumoslabs - No worries!  This is open source:  No need to apologize.  When you make your changes, be sure to check out [my comment](https://github.com/traviscrawford/spark-dynamodb/pull/8#issuecomment-252777066) which points to another Spark 2.0 incompatibility.

I'd be happy to test once ready.  Cheers!

**Comment by timchan-lumoslabs** [comment]
@findchris interesting.. the Scala API still has the `DataFrameReader` returning a `DataFrame`.
https://spark.apache.org/docs/2.0.1/api/scala/index.html#org.apache.spark.sql.DataFrameReader

I'm pretty sure everything will work once I bump up the Spark version to 2.0.1.

**Comment by timchan-lumoslabs** [comment]
@traviscrawford I've made my changes and am ready to merge this with your master. I think I made a mess of things with git. I don't know what to make of the rebase error. On my end, it says my master can merge with your master without conflicts.

**Comment by findchris** [comment]
Thanks @timchan-lumoslabs.

@traviscrawford - I'm happy to verify in production on Spark 2.0 when you're ready.

**Comment by findchris** [comment]
@traviscrawford - Even if you can merge this into a feature branch (based on your `master`), I can build and test if needed.  I am anxious to give this a run on Spark 2.0 in production.  Thanks!

**Comment by timchan-lumoslabs** [comment]
@findchris If you'd like to test, you could just build my master locally (https://github.com/timchan-lumoslabs/spark-dynamodb) and play with that? That way, you don't have to wait for @traviscrawford.

**Comment by findchris** [comment]
@timchan-lumoslabs Did you try to build with maven?  I tried, and got this:
```
[ERROR] /Users/chris/code/spark-dyanmodb-timechan/src/main/scala/com/github/traviscrawford/spark/dynamodb/DefaultSource.scala:43: error: type mismatch;
[ERROR]  found   : org.apache.spark.sql.SQLContext
[ERROR]  required: org.apache.spark.sql.SparkSession
[ERROR]       maybeEndpoint = parameters.get("endpoint"))(sqlContext)
```
...which points [here](https://github.com/timchan-lumoslabs/spark-dynamodb/blob/master/src/main/scala/com/github/traviscrawford/spark/dynamodb/DefaultSource.scala#L43).

**Comment by timchan-lumoslabs** [comment]
I must have been sloppy. It built at one point and then I must have messed things up while rebasing/merging/updating the PR. Thanks for catching that @findchris

**Comment by findchris** [comment]
@timchan-lumoslabs, are you working to resolve?

**Comment by traviscrawford** [comment]
Taking a look... will see if I can get this all synced up with master.

**Comment by timchan-lumoslabs** [comment]
@findchris I've made an update that fixes the build error.

**Comment by findchris** [comment]
@traviscrawford I'm building now and will run in production to test.
@timchan-lumoslabs Build succeeded, thanks.  Now to test.

**Comment by traviscrawford** [comment]
Tests passed locally - thanks for the contribution @timchan-lumoslabs!

@findchris Please let me know how testing in production works for you. If things look good I will publish a new release to Maven Central.

**Comment by timchan-lumoslabs** [comment]
@findchris I'm excited to see if these changes work for you.

**Comment by findchris** [comment]
@timchan-lumoslabs @traviscrawford - My job succeeded, so ostensibly we are good to merge this 👍

**Comment by timchan-lumoslabs** [comment]
@findchris @traviscrawford great collaboration on this PR!

**Comment by traviscrawford** [comment]
This has been published to maven central as release `0.0.6`.

**Comment by findchris** [comment]
Verified my code works with `0.0.6` release.  Thanks @traviscrawford  and @timchan-lumoslabs!  🎉
